### PR TITLE
Add Brick and test

### PR DESCRIPTION
**Pull request by mar-celine**

### CODE REVIEW

File: tests/Unit/Domain/DomainCreators/Test_Brick.cpp [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+           {Direction<3>::lower_zeta()},
+           {Direction<3>::upper_zeta()}}},
+      make_vector(make_coordinate_map<Frame::Logical, Frame::Inertial>(
+          affine_map_3d{affine_map{-1., 1., lower_bound[0], upper_bound[0]},
+                        affine_map{-1., 1., lower_bound[1], upper_bound[1]},
+                        affine_map{-1., 1., lower_bound[2], upper_bound[2]}})));
+}
+
+void test_periodic_brick(
+    const DomainCreators::Brick& brick,
```
Comment (mar-celine) [inline tests/Unit/Domain/DomainCreators/Test_Brick.cpp]: need to test other periodic boundaries as well (only periodic in x, periodic in x and y, etc)

File: src/Domain/DomainCreators/Brick.hpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+namespace DomainCreators {
+
+/// Create a 3D Domain consisting of a single Block.
+class Brick : public DomainCreator<3, Frame::Inertial> {
+ public:
+  struct LowerBound {
+    using type = std::array<double, 3>;
+    static constexpr OptionString_t help = {
+        "Sequence of [x,y] for lower bounds."};
```
Comment (kidder) [inline src/Domain/DomainCreators/Brick.hpp]: x,y,z

File: src/Domain/DomainCreators/Brick.hpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  };
+
+  struct UpperBound {
+    using type = std::array<double, 3>;
+    static constexpr OptionString_t help = {
+        "Sequence of [x,y] for upper bounds."};
```
Comment (kidder) [inline src/Domain/DomainCreators/Brick.hpp]: x,y,z

File: src/Domain/DomainCreators/Brick.hpp [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  };
+  struct IsPeriodicIn {
+    using type = std::array<bool, 3>;
+    static constexpr OptionString_t help = {
+        "Sequence for [x,y], true if periodic."};
```
Comment (kidder) [inline src/Domain/DomainCreators/Brick.hpp]: x,y,z

File: src/Domain/DomainCreators/CMakeLists.txt [hunk 5]
```diff
@@ -6,6 +6,7 @@ set(LIBRARY DomainCreators)
 set(LIBRARY_SOURCES
     Interval.cpp
     Rectangle.cpp
+    Brick.cpp
```
Comment (kidder) [inline src/Domain/DomainCreators/CMakeLists.txt]: alphabetical

**Review by kidder — CHANGES_REQUESTED** [review]
There is no reason you need to have the Rectangle commits in this PR as they are independent.

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
make same style changes that you're making to rectangle

File: src/Domain/DomainCreators/Brick.cpp [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+namespace DomainCreators {
+
+Brick::Brick(
+    const typename LowerBound::type lower_xyz,
+    const typename UpperBound::type upper_xyz,
+    const typename IsPeriodicIn::type is_periodic_in_xyz,
+    const typename InitialRefinement::type initial_refinement_level_xyz,
+    const typename InitialGridPoints::type initial_number_of_grid_points_in_xyz,
+    const OptionContext& /*context*/) noexcept
+    : lower_xyz_(lower_xyz),
```
Comment (kidder) [inline src/Domain/DomainCreators/Brick.cpp]: std::move the things you moved in rectangle

File: src/Domain/DomainCreators/CMakeLists.txt [hunk 7]
```diff
@@ -5,6 +5,7 @@ set(LIBRARY DomainCreators)
 
 set(LIBRARY_SOURCES
     Interval.cpp
```
Comment (kidder) [inline src/Domain/DomainCreators/CMakeLists.txt]: ABCDEFGHIJKLMNOPQRSTUVWXYZ

**Review by kidder — CHANGES_REQUESTED** [review]

File: src/Domain/DomainCreators/Brick.cpp [hunk 8]
```diff
@@ -25,12 +25,12 @@ Brick::Brick(
     const typename InitialRefinement::type initial_refinement_level_xyz,
     const typename InitialGridPoints::type initial_number_of_grid_points_in_xyz,
     const OptionContext& /*context*/) noexcept
-    : lower_xyz_(lower_xyz),
-      upper_xyz_(upper_xyz),
-      is_periodic_in_xyz_(is_periodic_in_xyz),
-      initial_refinement_level_xyz_(initial_refinement_level_xyz),
+    : lower_xyz_(std::move(lower_xyz)),
```
Comment (kidder) [inline src/Domain/DomainCreators/Brick.cpp]: remove consts from the variables you are moving out of

**Review by kidder — CHANGES_REQUESTED** [review]

File: tests/Unit/Domain/DomainCreators/Test_Brick.cpp [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+  test_factory_creation<DomainCreator<3, Frame::Inertial>>(
+      "  Brick:\n"
+      "    LowerBound: [0,0,0]\n"
+      "    UpperBound: [1,2,3]\n"
+      "    IsPeriodicIn: [True,False,True]\n"
+      "    InitialGridPoints: [3,4,3]\n"
+      "    InitialRefinement: [2,3,2]\n");
+}
+
+// [[OutputRegex, index = 1]]
```
Comment (nilsdeppe) [inline tests/Unit/Domain/DomainCreators/Test_Brick.cpp:177]: doesn't the ASSERT print more than this? i.e. I would expect it to print the condition that failed as well. If so, you should check that as part of the regex. The same applies for the test below, and in rectangle and interval.

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

File: tests/Unit/Domain/DomainCreators/Test_Brick.cpp (hunk 9 above)
Comment (nilsdeppe) [reply tests/Unit/Domain/DomainCreators/Test_Brick.cpp:177]: Changed my mind, function will disappear anyway.

**Review by nilsdeppe — DISMISSED** [review]

**Review by kidder — DISMISSED** [review]

**Review by kidder — APPROVED** [review]
